### DISCUSSION TITLE
Access the thread from its parent message

### DISCUSSION THREAD

**Discussion by PliskinGH** [Rapptz/discord.py#9664] [Q&A] [answered]
Hello,

Is there a way to retrieve a pre existing thread for a given starting message in a text channel?
Py-cord has the "thread" attribute in the Message class. But discord.py doesn't.

I can use get_thread or get_channel to retrieve the thread from its id, but is there any guarantee that the thread id is the same as the message id, and that it will always stay the same?

Thanks!

**Comment by LostLuma** [comment, marked as answer]
Using `get_thread` is indeed what you'll want to use in this situation at this time.  
The message and thread ID for a public thread started from a message being the same is also always going to be the same, this is even noted in the official documentation: [`The created thread and the message it was started from will share the same id. `](https://discord.com/developers/docs/topics/threads#public-private-threads)

**Reply by PliskinGH** [reply]
Thanks, I will do so.

I think it would be nice if the library provided a direct way though. There is a starting_message property in Thread, but not the other way around.
